src/ping.c

```diff
@@ -631,28 +661,13 @@ static int ping_read(void) /* {{{ */
       continue;
     }
 
-    /* Calculate average. Beware of division by zero. */
-    if (pkg_recv == 0)
-      latency_average = NAN;
-    else
-      latency_average = latency_total / ((double)pkg_recv);
-
-    /* Calculate standard deviation. Beware even more of division by zero. */
-    if (pkg_recv == 0)
-      latency_stddev = NAN;
-    else if (pkg_recv == 1)
-      latency_stddev = 0.0;
-    else
-      latency_stddev = sqrt(((((double)pkg_recv) * latency_squared) -
-                             (latency_total * latency_total)) /
-                            ((double)(pkg_recv * (pkg_recv - 1))));
-
     /* Calculate drop rate. */
     droprate = ((double)(pkg_sent - pkg_recv)) / ((double)pkg_sent);
 
-    submit(hl->host, "ping", latency_average);
-    submit(hl->host, "ping_stddev", latency_stddev);
-    submit(hl->host, "ping_droprate", droprate);
+    submit_gauge(hl->host, "ping_droprate", droprate);
+    submit_distribution(
+        hl->host, "ping_distribution_latency",
+        dist_latency); // dist_latency will be destroyed in submit_distribution
```

Comment:
Just to let you know:
I could argue if dist_latency is really destroyed in submit_distribution. As you can see, in submit_distribution you create a temporary metric with this distribution, nextly you append it to the metric family and at the end you free the metric_family. However I think there is a bug inside metric_reset function which metric_family_free calls and there is a place where a distribution_destroy call is missing. Also I think metric_family_metric_append should clone the distribution instead using the same.

I will fix these bugs in another pull request